src/main/java/org/jitsi/srtp/crypto/SrtpCipherGcm.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright @ 2015 - present 8x8, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.srtp.crypto;
+
+import javax.crypto.*;
+import javax.crypto.spec.*;
+import java.security.*;
+
+/**
+ * SRTP encryption in GCM mode.
+ */
+public class SrtpCipherGcm
+    extends SrtpCipher
+{
+    private static final byte[] zeroIv = new byte[12];
+    private final GCMParameterSpec param;
+
+    private SecretKeySpec key = null;
+
+    public SrtpCipherGcm(Cipher cipher, int authTagBits)
```

Comment:
I believe the GCM implementations themselves should be enforcing authTagBits lengths (only certain values are valid -- 128, 96, and as I recall 64 and 32 with some serious restrictions on key lifetime).  I'm willing to leave it up to the cipher implementation to enforce this, unless you think this is a bad idea?

Reply:
Well, rfc7714#section-13.2 specifically mandates 128bit, so having the parameter here and then only forcing a specific value  only in the OpenSSL cipher is at least confusing and not enforcing it in every case.
Maybe instead of removing the parameter it would be best to just check here for `authTagBits == 128`. Then the enforcement is also valid for SunJCE and BouncyCastle.